scripts/reactor_deployment.py

```diff
@@ -0,0 +1,496 @@
+# These functions outline different reactor deployment algorithms,
+# and metrics for analyzing the final product.
+
+import pandas as pd
+import numpy as np
+import math  # for rounding ceiling and floor
+from datetime import datetime  # for random seed generation
+
+
+# # # # # # # # # # # # Constituent Functions # # # # # # # # # # #
+
+
+def direct_decom(df, ar_dict):
+    """
+    This function assumes that every time a reactor model is decommissioned it
+    is replaced by a new version of itself.
+
+    Parameters
+    ----------
+    df: pandas dataframe
+        The dataframe of capacity information
+    ar_dict: dictionary
+        A dictionary of reactors with information of the form:
+        {reactor: [Power (MWe), capacity_factor (%), lifetime (yr)]}
+    """
+
+    # now we are going to note when reactors are decommissioned
+    for reactor in ar_dict.keys():
+        # create a decommissioning column
+        df[f'{reactor}Decom'] = 0
+        for year in range(len(df['Year'])):
+            decom_year = year + ar_dict[reactor][2]
+            if decom_year >= len(df['Year']):
```

Comment:
Since you use `len(df['Year'])` twice here, it might be worth having a variable for it like `num_years` (something that describes what it is).

Reply:
good idea!